### PR TITLE
docs: gatekeeper for 2.2

### PR DESCRIPTION
**Pull request by cwyl02**
NOTE: You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4112.s3-website-us-west-2.amazonaws.com/

## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-78966

## Description of changes being made

policy control / gatekeeper docs for 2.2

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Create new PRs against `develop`, or `main` for hotfixes to production.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

### CODE REVIEW

**Review by vacas5 — APPROVED** [review]

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+The Gatekeeper repository includes a [library of policies][gatekeeper-library] to replace Pod Security Policies which we will use in the following tutorials.
+
+### Prevent privileged pods
+
+#### Define the ConstraintTemplate
+
+Create the privileged pod policy constraint template `k8spspprivilegedcontainer` by running the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper-library/master/library/pod-security-policy/privileged-containers/template.yaml
```
Comment (takirala) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:42]: i dont like that we are using `../master/..` in the path as it is a moving depedency. looks like the repo does not have any tags so i guess this is fine.

**Review by takirala — APPROVED** [review]
cc @mikolajb can you glance over this just to make sure that the `3.7.0` bump doesnt change anything (i dont think it should, but just in case).

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+#### Define the Constraint
+
+Constraints are then used to inform Gatekeeper that the admin wants a ConstraintTemplate to be enforced, and how.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:47]:
```suggestion
Constraints are then used to inform Gatekeeper that the admin wants to enforce a ConstraintTemplate, and how.
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+Create the host path volume policy constraint template `k8spsphostfilesystem` by running the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper-library/master/library/pod-security-policy/host-filesystem/template.yaml
+```
+
+#### Define the Constraint
+
+Constraints are then used to inform Gatekeeper that the admin wants a ConstraintTemplate to be enforced, and how.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:81]:
```suggestion
Constraints are then used to inform Gatekeeper that the admin wants to enforce a ConstraintTemplate, and how.
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  parameters:
+    allowedHostPaths:
+    - readOnly: true
+      pathPrefix: "/foo"
+EOF
+```
+
+#### Test that the constraint is enforced
+
+Try to create a pod that mounts a disallowed host path by running the following command:
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:105]:
```suggestion
Create a pod that mounts a disallowed host path by running the following command:
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+You should see the following output:
+
+```bash
+Error from server ([denied by psp-host-filesystem] HostPath volume {"hostPath": {"path": "/tmp", "type": ""}, "name": "cache-volume"} is not allowed, pod: nginx-host-filesystem. Allowed path: [{"readOnly": true, "pathPrefix": "/foo"}]): error when creating "STDIN": admission webhook "validation.gatekeeper.sh" denied the request: [denied by psp-host-filesystem] HostPath volume {"hostPath": {"path": "/tmp", "type": ""}, "name": "cache-volume"} is not allowed, pod: nginx-host-filesystem. Allowed path: [{"readOnly": true, "pathPrefix": "/foo"}]
+```
+
+#### Test that the constraint allows allowed host paths
+
+Try to create a pod that mounts an allowed host path by running the following command:
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:138]:
```suggestion
Create a pod that mounts an allowed host path by running the following command:
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+- You must have a properly deployed and running cluster. For information about deploying Kubernetes with default settings on different types of infrastructures, see the [Choose Infrastructure][choose-infrastructure] topic.
+
+- If you install Kommander with a custom configuration, make sure you enabled gatekeeper.
+
+## Use Gatekeeper
+
+Gatekeeper uses the [OPA Constraint Framework][opa-constraints] to describe and enforce policy. Before you can define a constraint, you must first define a `ConstraintTemplate`, which describes both the [`Rego`][opa-rego] (a powerful query language) that enforces the constraint and the schema of the constraint. The schema of the constraint allows an admin to fine-tune the behavior of a constraint, much like arguments to a function.
+
+The Gatekeeper repository includes a [library of policies][gatekeeper-library] to replace Pod Security Policies which we will use in the following tutorials.
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:33]:
```suggestion
The Gatekeeper repository includes a [library of policies][gatekeeper-library] to replace Pod Security Policies which you will use in the following tutorials.
```

**Review by gvasquezvargas — COMMENTED** [review]
Just some comments on the language. Don't hesitate to provide feedback if needed! Thanks!

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+Create the privileged pod policy constraint `psp-privileged-container` by running the following command:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper-library/master/library/pod-security-policy/privileged-containers/samples/psp-privileged-container/constraint.yaml
+```
+
+#### Test that the constraint is enforced
+
+Try to create a privileged pod by running the following command:
```
[the remark this answers is not in the document]
Comment (cwyl02) [reply pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md]: done

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md (hunk 4 above)
Comment (cwyl02) [reply pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md]: done

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md (hunk 6 above)
Comment (cwyl02) [reply pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md]: done

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md (hunk 5 above)
Comment (cwyl02) [reply pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md]: done

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+excerpt: Learn how to enforce policies using Gatekeeper
+beta: false
+enterprise: false
+---
+
+<!-- markdownlint-disable MD024 -->
+
+[Gatekeeper][gatekeeper] is the policy controller for Kubernetes, allowing organizations to enforce configurable policies using the [Open Policy Agent][opa], a policy engine for Cloud Native environments hosted by CNCF as a graduated-level project.
+
+This tutorial describes how to use Gatekeeper to enforce policies by rejecting non-compliant resources. Specifically, this tutorial describes two constraints as a way to show how to use Gatekeeper as an alternative to [Pod Security Policies][psp]:
```
Comment (JohnB64-JBV) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:15]:
```suggestion
This tutorial describes how to use Gatekeeper to enforce policies by rejecting non-compliant resources. Specifically, this tutorial describes two constraints as a way to use Gatekeeper as an alternative to [Pod Security Policies][psp]:
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+- [Prevent mounting host path volumes](#prevent-host-path-volumes)
+
+## Before you begin
+Before starting this tutorial, verify the following:
+
+- You must have access to a Linux, macOS, or Windows computer with a supported operating system version.
+
+- You must have a properly deployed and running cluster. For information about deploying Kubernetes with default settings on different types of infrastructures, see the [Choose Infrastructure][choose-infrastructure] topic.
+
+- If you install Kommander with a custom configuration, make sure you enabled gatekeeper.
```
Comment (JohnB64-JBV) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:27]:
```suggestion
- If you install Kommander with a custom configuration, make sure you enabled Gatekeeper.
```

File: pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+EOF
+```
+
+You should see the following output:
+
+```bash
+Error from server ([denied by psp-host-filesystem] HostPath volume {"hostPath": {"path": "/tmp", "type": ""}, "name": "cache-volume"} is not allowed, pod: nginx-host-filesystem. Allowed path: [{"readOnly": true, "pathPrefix": "/foo"}]): error when creating "STDIN": admission webhook "validation.gatekeeper.sh" denied the request: [denied by psp-host-filesystem] HostPath volume {"hostPath": {"path": "/tmp", "type": ""}, "name": "cache-volume"} is not allowed, pod: nginx-host-filesystem. Allowed path: [{"readOnly": true, "pathPrefix": "/foo"}]
+```
+
+#### Test that the constraint allows allowed host paths
```
Comment (JohnB64-JBV) [inline pages/dkp/kommander/2.2/security/policy-controls/gatekeeper/index.md:136]:
```suggestion
#### Test that the constraint allows the allowed host paths
```

**Review by JohnB64-JBV — CHANGES_REQUESTED** [review]

**Review by JohnB64-JBV — APPROVED** [review]

**Pull request merged**